prime-router/src/main/kotlin/transport/SoapTransport.kt

```diff
@@ -179,9 +182,44 @@ class SoapTransport(private val httpClient: HttpClient? = null) : ITransport {
                 ", Exception: ${t.localizedMessage}"
             context.logger.severe(msg)
             context.logger.severe(t.stackTraceToString())
-            actionHistory.setActionType(TaskAction.send_warning)
-            actionHistory.trackActionResult(msg)
-            RetryToken.allItems
+            // do some additional handling of the error here. if we are dealing with a 400 error, we
+            // probably don't want to retry, and we need to stop now
+            // if the error is a 500 we can do a retry, but both should probably throw a pager duty notification
+            when (t) {
+                is ClientRequestException -> {
+                    (t as ClientRequestException).let {
+                        context.logger.severe(
+                            "Received ${it.response.status.value}: ${it.response.status.description} " +
```

Comment:
So, I request that you also log the SOAP body. It often has the most important information about the error. This applies to the other exception cases.

Reply:
Wouldn't the SOAP body potentially have PII in it?

Reply:
The response body could potentially, if they log what we sent to them back at us, but I don’t think it typically would.